src/simple_http.rs

```diff
@@ -246,6 +246,65 @@ fn get_lines<R: BufRead>(reader: &mut R) -> Result<String, Error> {
     Ok(body)
 }
 
+/// Do some very basic manual URL parsing because the uri/url crates
+/// all have unicode-normalization as a dependency and that's broken.
+fn check_url(url: &str) -> Result<SocketAddr, Error> {
```

Comment:
Can we have a unit test please added as a separate patch before the first refactor patch to test `url()` - that way we know that the refactor does not introduce any logic changes.

Reply:
I was thinking about the same. But these tests here https://github.com/apoelstra/rust-jsonrpc/blob/e3d92e30e1a6f04cbd2278f05a41966f203a65a2/src/simple_http.rs#L400-L451

seemed to be covering the the new moved codes too.. So in theory we are having the same test coverage.. And as there wasn't (suppose to be) any changed logic, so I couldn't figure what new tests to write..
But its interesting that we didn't fail even when `self.tp.path` wasn't being set properly. Maybe that's the compiler telling us we need more tests?

Reply:
I added some basic assertions that we are setting the values correctly here.. https://github.com/apoelstra/rust-jsonrpc/pull/70/commits/9f10f71a2f51e32953d52d1265522b5a61b37e78